pkg/api/status_api_handlers.go

```diff
@@ -76,3 +77,206 @@ func (s *statusAPI) GetCephMonDump(ctx context.Context, req *emptypb.Empty) (*pb
+			ApplicationMetadata:               pool.ApplicationMetadata,
+			ReadBalance:                       pool.ReadBalance,
+		})
+	}
+
+	blocklistPb := make(map[string]*timestamppb.Timestamp, len(osdDump.Blocklist))
+	for ip, t := range osdDump.Blocklist {
+		goParsedTime, err := parseCustomTimestamp(t)
+		if err != nil {
+			return nil, err
+		}
+		blocklistPb[ip] = timestamppb.New(goParsedTime)
+	}
+
+	var osdXInfo []*pb.OsdDumpOsdXInfo
+	for _, osdX := range osdDump.OsdXinfo {
+		downStamp, err := parseCustomTimestamp(osdX.DownStamp)
+		if err != nil {
+			return nil, err
+		}
+		lastPurgedSnapsScrub, err := parseCustomTimestamp(osdX.LastPurgedSnapsScrub)
+		if err != nil {
+			return nil, err
+		}
+		osdXInfo = append(osdXInfo, &pb.OsdDumpOsdXInfo{
+			Osd:                  osdX.Osd,
+			DownStamp:            timestamppb.New(downStamp),
+			LaggyProbability:     osdX.LaggyProbability,
+			LaggyInterval:        osdX.LaggyInterval,
+			Features:             osdX.Features,
+			OldWeight:            osdX.OldWeight,
+			LastPurgedSnapsScrub: timestamppb.New(lastPurgedSnapsScrub),
+			DeadEpoch:            osdX.DeadEpoch,
+		})
+	}
+
+	response := pb.GetCephOsdDumpResponse{
+		Epoch:                  osdDump.Epoch,
+		Fsid:                   osdDump.Fsid,
+		Modified:               modifiedTimestamp,
+		Created:                createdTimestamp,
+		LastUpChange:           lastUpChange,
+		LastInChange:           lastInChange,
+		Flags:                  osdDump.Flags,
+		FlagsNum:               osdDump.FlagsNum,
+		FlagsSet:               osdDump.FlagsSet,
+		CrushVersion:           osdDump.CrushVersion,
+		FullRatio:              osdDump.FullRatio,
+		BackfillfullRatio:      osdDump.BackfillfullRatio,
+		NearfullRatio:          osdDump.NearfullRatio,
+		ClusterSnapshot:        osdDump.ClusterSnapshot,
+		PoolMax:                osdDump.PoolMax,
+		MaxOsd:                 osdDump.MaxOsd,
+		RequireMinCompatClient: osdDump.RequireMinCompatClient,
+		MinCompatClient:        osdDump.MinCompatClient,
+		RequireOsdRelease:      osdDump.RequireOsdRelease,
+		AllowCrimson:           osdDump.AllowCrimson,
+		Pools:                  osdDumpPools,
+
+		Osds:     osdDump.Osds,
+		OsdXinfo: osdXInfo,
+
+		PgUpmap:          osdDump.PgUpmap,
+		PgUpmapItems:     osdDump.PgUpmapItems,
+		PgUpmapPrimaries: osdDump.PgUpmapPrimaries,
+		PgTemp:           osdDump.PgTemp,
+		PrimaryTemp:      osdDump.PrimaryTemp,
+
+		Blocklist:           blocklistPb,
+		RangeBlocklist:      osdDump.RangeBlocklist,
+		ErasureCodeProfiles: osdDump.ErasureCodeProfiles,
+
+		RemovedSnapsQueue: osdDump.RemovedSnapsQueue,
+		NewRemovedSnaps:   osdDump.NewRemovedSnaps,
+		NewPurgedSnaps:    osdDump.NewPurgedSnaps,
+
+		CrushNodeFlags:   osdDump.CrushNodeFlags,
+		DeviceClassFlags: osdDump.DeviceClassFlags,
+		StretchMode:      osdDump.StretchMode,
+	}
+
+	return &response, nil
+}
+
+func parseCustomTimestamp(timestamp string) (time.Time, error) {
```

Comment:
is it makes sense to implement custom json unmarshaling instead?

Something like:
```go
type CephTime time.Time
func (receiver *CephTime) UnmarshalJSON(data []byte) error {
...
```
see https://stackoverflow.com/a/65275814/6065134